Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs

```diff
@@ -0,0 +1,478 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// Object that represents a cursor in 3D space controlled by gaze.
+    /// </summary>
+    public abstract class Cursor : MonoBehaviour, ICursor
+    {
+        /// <summary>
+        /// Enum for current cursor state
+        /// </summary>
+        public enum CursorStateEnum
+        {
+            /// <summary>
+            /// Useful for releasing external override.
+            /// See <c>CursorStateEnum.Contextual</c>
+            /// </summary>
+            None = -1,
+            /// <summary>
+            /// Not IsHandVisible
+            /// </summary>
+            Observe, 
+            /// <summary>
+            /// Not IsHandVisible AND not IsInputSourceDown AND TargetedObject exists
+            /// </summary>
+            ObserveHover, 
+            /// <summary>
+            /// IsHandVisible AND not IsInputSourceDown AND TargetedObject is NULL
+            /// </summary>
+            Interact,
```

Comment:
Yeah, it is. I think the way @paseb names the states, "Interact" means that the user's hand is visible with the finger in the ready state, so the user is in a "ready to interact." state. InteractHover is the equivalent for when the user's gaze is hovering over an object.

I don't object to renaming those if you have a better name for the states, but I don't have a better suggestion in mind.